lib/OpenQA/WebAPI/Controller/API/V1/Worker.pm

```diff
@@ -114,18 +115,42 @@ sub _register {
     # to still work on these jobs (which might be the case when the worker hasn't actually crashed but
     # just re-registered due to network issues)
     my %jobs_worker_says_it_works_on = map { ($_ => 1) } @$jobs_worker_says_it_works_on;
-    $worker->update({job_id => undef}) if _incomplete_previous_job(\%jobs_worker_says_it_works_on, $worker->job);
-    _incomplete_previous_job(\%jobs_worker_says_it_works_on, $_) for $worker->unfinished_jobs->all;
+    my $worker_id                    = $worker->id;
+    try {
+        $schema->txn_do(
+            sub {
+                $worker->update({job_id => undef})
+                  if _incomplete_previous_job(\%jobs_worker_says_it_works_on, $worker->job);
+                _incomplete_previous_job(\%jobs_worker_says_it_works_on, $_) for $worker->unfinished_jobs->all;
+            });
+    }
+    catch {
+        log_warning("Unable to incomplete/duplicate or reschedule jobs abandoned by worker $worker_id: $_");
```

Comment:
is this something you would foresee to happen but we do not want to leave the exception unhandled?

Reply:
These kind of errors were also previously catched in the scheduler code. So I assume they can happen (hence also the transaction) but it wouldn't make sense to fail the worker registration due to incompletion errors.